python/ray/data/_internal/execution/resource_manager.py

```diff
@@ -109,21 +113,48 @@ def update_usages(self):
         # And some computations are redundant. We should either remove redundant
         # computations or remove this method entirely and compute usages on demand.
         self._global_usage = ExecutionResources(0, 0, 0)
+        self._global_running_usage = ExecutionResources(0, 0, 0)
+        self._global_pending_usage = ExecutionResources(0, 0, 0)
         self._op_usages.clear()
+        self._op_running_usages.clear()
+        self._op_pending_usages.clear()
         self._downstream_fraction.clear()
         self._downstream_object_store_memory.clear()
 
         # Iterate from last to first operator.
         num_ops_so_far = 0
         num_ops_total = len(self._topology)
         for op, state in reversed(self._topology.items()):
-            # Update `self._op_usages`.
+            # Update `self._op_usages`, `self._op_running_usages`,
+            # and `self._op_pending_usages`.
             op_usage = op.current_processor_usage()
+            op_running_usage = op.running_processor_usage()
+            op_pending_usage = op.pending_processor_usage()
+
             assert not op_usage.object_store_memory
+            assert not op_running_usage.object_store_memory
+            assert not op_pending_usage.object_store_memory
             op_usage.object_store_memory = self._estimate_object_store_memory(op, state)
+            op_running_usage.object_store_memory = self._estimate_object_store_memory(
+                op, state
+            )
+            op_pending_usage.object_store_memory = self._estimate_object_store_memory(
+                op, state
+            )
```

Comment:
this will count `_estimate_object_store_memory` twice? 
We can only count the estimated object store memory usage to running usage for now. 
later we can separate the running and pending usage.

Reply:
Fair enough, and pending usage's object store memory is not used regardless. We should calculate its object store memory once we address #47125.